data-prepper-plugins/http-source-common/src/main/java/org/opensearch/dataprepper/http/codec/JsonCodec.java

```diff
@@ -47,21 +47,20 @@ public List<List<String>> parse(HttpData httpData, int maxSize) throws IOExcepti
         List<List<String>> jsonList = new ArrayList<>();
         final List<Map<String, Object>> logList = mapper.readValue(httpData.toInputStream(),
                 LIST_OF_MAP_TYPE_REFERENCE);
-        int size = OVERHEAD_CHARACTERS.length();
         List<String> innerJsonList = new ArrayList<>();
-        for (final Map<String, Object> log: logList) {
-            final String recordString = mapper.writeValueAsString(log);
-            int nextRecordLength = recordString.getBytes(Charset.defaultCharset()).length;
+        String recordString = mapper.writeValueAsString(logList.get(0));
```

Comment:
I'm not sure I follow this change from your previous commit. Why would index `0` be a special case?

Reply:
I can go back to previous commit, I thought this way, I can avoid the check inside the loop. I am OK, either way

Reply:
I found the previous commit to be clearer.